storm-core/src/jvm/org/apache/storm/trident/TridentState.java

```diff
@@ -23,18 +23,33 @@
 public class TridentState {
     TridentTopology _topology;
     Node _node;
-    
+
     protected TridentState(TridentTopology topology, Node node) {
         _topology = topology;
         _node = node;
     }
-    
+
     public Stream newValuesStream() {
         return new Stream(_topology, _node.name, _node);
     }
-    
+
     public TridentState parallelismHint(int parallelism) {
         _node.parallelismHint = parallelism;
         return this;
     }
+
+    public TridentState setCPULoad(Number load) {
```

Comment:
Can we do the same here too?  Have TridentState implement `ResourceDeclarer<TridentState>` and have these marked with `@Override`.